examples/usb_i2c_mctp.rs

```diff
@@ -0,0 +1,225 @@
+//! UART and I2C.
+//!
+//! The following topology is used:
+//!
+//! [HOST MACHINE] <--UART--> [USB_I2C_BRIDGE_DEVICE] <--I2C/SmBus--> [TARGET_ENDPOiNT]
+//!
+//! The host machine will issue a message to the USB_I2C_BRIDGE_DEVICE, which runs this app.
+//! The app then determines the end point target address based on the received packet header,
+//! and forwards this data to the target endpoint. As we are supporting MCTP, there are no
+//! I2C reads, only write. Thus, the endpoint must then master the bus and write a response
+//! back. This message is forwarded to the host again via UART.
+//!
+//! The host application must append a small packet header of the following format
+//! for any messages being send to this device.
+//!
+//! host_tx[0] = 0xAA // Preamble
+//! host_tx[1] = XX   // Endpoint target address
+//! host_tx[2] = YY   // MSB of 16bit data length
+//! host_tx[3] = ZZ   // LSB of 16bit data length
+//!
+//! For reception, the device (this app) first sends a packet header of the following format
+//!
+//! device_tx[0] = 0xBB // Preamble
+//! device_tx[1] = 0xFF // Unused
+//! device_tx[2] = YY   // MSB of 16bit data length
+//! device_tx[3] = ZZ   // LSB of 16bit data length
+//!
+//! Based on the data length, the host can read the next ((YY << 8) | ZZ )
+//! as data message.
+//!
+//! Required Kernel Configuration:
+//!
+//! This application requires that the kernel console buffer sizes are increased
+//! as well as i2c-master-slave driver buffers. This is because we are doing transfers
+//! of sizes greater than what the upstream kernel is allowed to do.
+//!
+//! The following reference can be used to prepare the kernel for the uart/console buffers.
+//! The size specified here must be >= RX_BUF_LEN (as specified below).
+//!
+//! ```
+//! diff --git a/capsules/core/src/virtualizers/virtual_uart.rs b/capsules/core/src/virtualizers/virtual_uart.rs
+//! index 0d39fe024..59e0435a1 100644
+//! --- a/capsules/core/src/virtualizers/virtual_uart.rs
+//! +++ b/capsules/core/src/virtualizers/virtual_uart.rs
+//! @@ -54,7 +54,7 @@ use kernel::hil::uart;
+//!  use kernel::utilities::cells::{OptionalCell, TakeCell};
+//!  use kernel::ErrorCode;
+//!
+//! -pub const RX_BUF_LEN: usize = 64;
+//! +pub const RX_BUF_LEN: usize = 132;
+//!
+//!  pub struct MuxUart<'a> {
+//!      uart: &'a dyn uart::Uart<'a>,
+//! ```
+//!
+//! ```
+//! diff --git a/capsules/core/src/virtualizers/virtual_uart.rs b/capsules/core/src/virtualizers/virtual_uart.rs
+//! index 0d39fe024..59e0435a1 100644
+//! --- a/capsules/core/src/virtualizers/virtual_uart.rs
+//! +++ b/capsules/core/src/virtualizers/virtual_uart.rs
+//! @@ -54,7 +54,7 @@ use kernel::hil::uart;
+//!  use kernel::utilities::cells::{OptionalCell, TakeCell};
+//!  use kernel::ErrorCode;
+//!
+//! -pub const RX_BUF_LEN: usize = 64;
+//! +pub const RX_BUF_LEN: usize = 132;
+//!
+//!  pub struct MuxUart<'a> {
+//!      uart: &'a dyn uart::Uart<'a>,
+//! ```
```

Comment:
This is just a copy of the section above right?

Reply:
aw crap yea, i copied one of the wrong diffs

Reply:
Updated with the correct 2 diffs